frontend/cypress/integration/common/waypoint.ts

```diff
@@ -61,6 +62,40 @@ const waitForBookinfoWaypointTrafficGeneratedInGraph = (maxRetries = 30, retryCo
   });
 };
 
+const isSyncedOrIgnored = (status: string | undefined): boolean => {
+  return status?.toLowerCase() === 'synced' || status?.toLowerCase() === 'ignored';
+};
+
+const proxyStatusHealthy = ({ proxyStatus }: Pod): boolean => {
+  return (
+    isSyncedOrIgnored(proxyStatus?.CDS) &&
+    isSyncedOrIgnored(proxyStatus?.EDS) &&
+    isSyncedOrIgnored(proxyStatus?.LDS) &&
+    isSyncedOrIgnored(proxyStatus?.RDS)
+  );
+};
+
+const waitForHealthyWaypoint = (maxRetries = 30, retryCount = 0): void => {
```

Comment:
this could include the namespace and the waypoint to be used in the api call and reused by other waypoints as well